delta_backend/src/ConversionChecker.py

```diff
@@ -85,25 +93,121 @@ def convertData(self, expressionType, expressionRule, fieldName, fieldValue):
                     expressionRule, fieldName, fieldValue, self.summarise, self.report_unexpected_exception
                 )
             case _:
-                return "Schema expression not found! Check your expression type : " + expressionType
+               return "Schema expression not found! Check your expression type : " + expressionType
+    
+    # Utility function for logging errors
+    def _log_error(self, fieldName, fieldValue, e, code=ExceptionMessages.RECORD_CHECK_FAILED):
```

Comment:
Do we always check `self.report_unexpected_exception` before calling `_log_error`? If so, could we check it once here instead?